internal/worker/bootstrap/manifold.go

```diff
@@ -58,6 +58,10 @@ type BootstrapAddressFinderGetter func(providerFactory providertracker.ProviderF
 // machine provisioned.
 type SetMachineProvisionedFunc func(context.Context, AgentPasswordService, MachineService, instancecfg.StateInitializationParams, agent.Config) error
 
+// FinaliseControllerNodeFunc is the function that is used to set the finish
+// setting up the controller node.
+type FinaliseControllerNodeFunc func(context.Context, ControllerNodeService, agent.Config) error
```

Comment:
We can reuse the `SetMachineProvisionedFunc` here. Just rename to fit both usecases.